### PR TITLE
Fix the report workflow to work with ci-group

### PR DESCRIPTION
**Pull request by zelinh**
### Description
Fix the report workflow to work with ci-group

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/4943

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/manifests/test_manifest.py [hunk 1]
```diff
@@ -31,6 +31,7 @@ class TestManifest(ComponentManifest['TestManifest', 'TestComponents']):
                 - with-less-security
               additional-cluster-configs:
                 - key : value
+              ci-group: 6
```
Comment (peterzhuamazon) [inline src/manifests/test_manifest.py:34]: Why hardcode here?

**Review by peterzhuamazon — CHANGES_REQUESTED** [review]

File: tests/tests_report_workflow/test_test_report_runner_dashboards.py [hunk 2]
```diff
@@ -51,14 +51,29 @@ def test_generate_file(self, report_args_mock: MagicMock) -> None:
...
+        report_args_mock.test_run_id = 123
+        report_args_mock.base_path = self.DATA_DIR
+        report_args_mock.test_type = "integ-test"
+        report_args_mock.release_candidate = "100"
+
+        test_report_runner = TestReportRunner(report_args_mock, self.TEST_MANIFEST_OPENSEARCH_DASHBOARDS)
+        test_report_runner_data = test_report_runner.update_data()
+
+        self.assertEqual(len(test_report_runner_data["components"]), 9)
+        self.assertEqual(test_report_runner_data["components"][0]["name"], "OpenSearch-Dashboards-ci-group-1")
```
Comment (gaiksaya) [inline tests/tests_report_workflow/test_test_report_runner_dashboards.py:75]: Can you also add assertions for all ci-groups? Until 9?

File: src/manifests/test_manifest.py (hunk 1 above)
Comment (gaiksaya) [reply src/manifests/test_manifest.py:34]: Looks like an example of the schema

Comment (zelinh) [reply src/manifests/test_manifest.py:34]: This is the schema for the test manifest. All these are commented out. Basically showing the structure and format for the test manifest.

File: tests/tests_report_workflow/test_test_report_runner_dashboards.py (hunk 2 above)
Comment (zelinh) [reply tests/tests_report_workflow/test_test_report_runner_dashboards.py:75]: Already assert the total number of components in the report manifest though. I would add assertions for all of them.

**Review by peterzhuamazon — APPROVED** [review]
Thanks.

**Pull request merged**